File: src/Services/RecruitmentService/Recruitment.API/DTOs/CandidateApplicationProgressDto.cs
namespace Recruitment.API.DTOs;

public sealed class CandidateApplicationProgressDto
{
    public CandidateProgressDto Progress { get; set; } = null!;
    public RecruitmentProcessDto Process { get; set; } = null!;
    public List<InterviewScheduleDto> Interviews { get; set; } = [];
}

File: src/Web/WebSPA/src/app/core/models/candidate-application-progress-dto.ts
import { CandidateProgressDto } from './candidate-progress-dto';
import { InterviewScheduleDto } from './interview-schedule-dto';
import { RecruitmentProcessDto } from './recruitment-process-dto';

export interface CandidateApplicationProgressDto {
  readonly progress: CandidateProgressDto;
  readonly process: RecruitmentProcessDto;
  readonly interviews: readonly InterviewScheduleDto[];
}
